src/shared/cpu-set-util.c

```diff
@@ -34,6 +34,46 @@ char* cpu_set_to_string(const CPUSet *a) {
         return TAKE_PTR(str) ?: strdup("");
 }
 
+char *cpu_set_to_range(const CPUSet *set) {
+        unsigned range_start = 0, range_end;
+        _cleanup_free_ char *str = NULL;
+        size_t allocated = 0, len = 0;
+        bool in_range = false;
+        int r;
+
+        for (unsigned i = 0; i < set->allocated * 8; i++) {
+                bool b = CPU_ISSET_S(i, set->allocated, set->set);
+
+                if (b) {
+                        if (in_range)
+                                range_end++;
+                        else {
+                                range_start = range_end = i;
+                                in_range = true;
+                        }
+                } else if (!b && in_range) {
+                        in_range = false;
+
+                        if (!GREEDY_REALLOC(str, allocated, len + 2 + 2 * DECIMAL_STR_MAX(unsigned)))
+                                return NULL;
+
+                        r = sprintf(str + len, len > 0 ? " %d-%d" : "%d-%d", range_start, range_end);
```

Comment:
hmm, this would mean if only cpu 5 is marked we'd output `5-5` for it, no? I think it would be nicer to output just `5` in that case, i.e. handle the `range_start == range_end` case special

Reply:
i see this has been fixed in a later commit in the same PR, so ignore me on this one